fuse_models/src/odometry_2d_publisher.cpp

```diff
@@ -446,15 +448,25 @@ void Odometry2DPublisher::publishTimerCallback(const ros::TimerEvent& event)
 
   if (params_.publish_tf)
   {
+    auto frame_id = odom_output.header.frame_id;
+    auto child_frame_id = odom_output.child_frame_id;
+
+    if (params_.invert_tf)
+    {
+      pose = pose.inverse();
+      std::swap(frame_id, child_frame_id);
+    }
+
     geometry_msgs::TransformStamped trans;
-    trans.header = odom_output.header;
-    trans.child_frame_id = odom_output.child_frame_id;
+    trans.header.stamp = odom_output.header.stamp;
+    trans.header.frame_id = frame_id;
+    trans.child_frame_id = child_frame_id;
     trans.transform.translation.x = pose.getX();
     trans.transform.translation.y = pose.getY();
     trans.transform.translation.z = odom_output.pose.pose.position.z;
     trans.transform.rotation = tf2::toMsg(pose.getRotation());
 
-    if (params_.world_frame_id == params_.map_frame_id)
+    if (frame_id == params_.map_frame_id)
```

Comment:
And then here as well
```
if (!params_.invert_tf && params_.world_frame_id == params_.map_frame_id)
```

Reply:
Done, but in the code above I still `std::swap` the frames `if (params_.invert_tf)`, since I believe that makes the implementation simpler.

Reply:
Yeah, sorry. That swap makes complete sense. But this is much easier to read now. Thank you.